### ISSUE TITLE
Improve Auto Trigger to work more than just the initial message

### ISSUE THREAD

**Issue by TypicallyShad** [modmail-dev/Modmail#3162]
### Is your feature request related to a problem? Please elaborate.

It seems that autotrigger has been purposely made to work only on the first initial message that is sent by the user. This isn't practical as most users begin a ticket with a greeting, such as hello rather than getting straight into their question. Most of my users also have more than one question they ask, which autotrigger can't cover. 

### Describe the solution you'd like

I would like for autotrigger to be able to be triggered for any messages that is sent in a thread, not just the initial starting message.

### Does your solution involve any of the following?

- [ ] Logviewer
- [x] New config option

### Describe alternatives you've considered

There isn't really alternative solutions, however, I assume it is possible with a plugin? 

### Who will this benefit?

I believe this will benefit a large portion of users who perhaps run technology groups or where autotrigger is commonly used for automatic help.

**Comment by Jerrie-Aries** [comment]
> That makes sense. I think this can be implemented without breaking the core functionality of the bot.

Yep, it wouldn't really break the core but the outcome could be ugly with the `auto` thing.

**Simple example:**
If we have an autotrigger keyword `I need help` which is pointed to `reply What help do you need?`, the bot would be triggered with any messages that contain the `I need help` even at the end of conversation with a message like `Next time I need help I will DM the bot again.`. That would not be something that we want.

**Comment by TypicallyShad** [comment]
I agree with that, in place, we could also ensure the system doesn't keep repeating the same auto trigger for similar phrases.

**Comment by Jerrie-Aries** [comment]
If you're talking about an autotrigger that could only be triggered with specific/unique keywords (which is basically like a command except without prefix and in DMs), that could be done with a plugin.

If you want you can try requesting the plugin in `#plugin-requests` channel in the [support server](https://discord.gg/zmdYe3ZVHG). Maybe someone can make one.✌

**Comment by StephenDaDev** [comment]
Thank you for your suggestion! The Modmail project thrives through implementing the features most desired by our community. However, we always need to maintain the tight balance of user friendliness, importantance (or in other words, bloat prevention), and stability.

I have reviewed this, and discussed this internally many times. The feature is a good idea in concept, but there would be a number of challenges to fully implement it effectively:
1. It would be necessary to set a limit on how many messages into a thread cause an autotrigger to work. Otherwise, many threads would end in an endless loop of improper autotrigger actions.
2. It would cause friction for the staff teams of servers in attempting to develop autotriggers that do not have any substantial amount of false positives.
3. It would take significant effort to get all of this working, for minimal reward (see below.)

Additionally, with the Modmail project being used in thousands of servers it remains important to avoid bloat, and see if there is another way to address the issue actually resulting in a feature request. Earlier this month, in v4.1.0, we introduced the threadmenu system, which allows server staff teams to develop a customized menu, with nesting functionality to address the situation of routing threads more effectively, regardless of the wording used by the recipient. We feel that this is a better solution for a majority of the users of the bot.

As Jerrie-Ares pointed out above, you can add this functionality with a plugin if you still wish to have it, it will just not have core-level implementation.
